radix-engine/src/vm/native_vm.rs

```diff
@@ -216,3 +235,19 @@ impl<C: VmInvoke + Clone> NativeVmExtension for OverridePackageCode<C> {
         }
     }
 }
+
+/// This function catches panic unwinds in the callbacks passed to it. If the crate is build with
+/// no-std then this function will not be able to catch the panics since [`catch_unwind`] is in the
+/// Rust standard library and not available to no_std:
+/// https://github.com/rust-lang/rfcs/issues/2810
+///
+/// [`catch_unwind`]: std::panic::catch_unwind
+fn catch_unwind<F: FnOnce() -> R, R>(f: F) -> Result<R, Box<dyn core::any::Any + Send + 'static>> {
```

Comment:
Note: we can't catch panics when building the engine with no-std since `std::panic::catch_unwind` is defined in the standard library. I'm mentioning this since one of our CI jobs runs the radix-engine-tests with no-std.

Reply:
Would prefer for this to just be more explicit. Rather than a call to `catch_unwind` which may or may not catch panics better to call `catch_unwind` if `std` and a direct call if `no_std`

Reply:
Agree that it's nicer. I've made that change.